src/logger.cr

```diff
@@ -7,47 +7,59 @@ class Logger
   alias FilterType = Filter | Severity | (Entry -> Bool)
   alias EmitterType = Emitter | (Entry -> Nil)
 
-  module Base
-    abstract def component : String
-    abstract def filter : FilterType?
-    abstract def emitter : EmitterType?
-
-    def log(entry : Entry)
-      case ff = filter
-      when Filter, Proc(Entry, Bool)
-        return unless ff.call(entry)
-      when Severity
-        return unless entry.severity >= ff
-      end
-      emitter.try &.call(entry)
-    end
+  property component : String
+  property filter : FilterType?
+  property emitters : Array(EmitterType)
 
-    {% for level in Severity.constants %}
-      def {{ level.downcase.id }}(*, time = Time.now, line_number = __LINE__, filename = __FILE__, &message : -> String)
-        log Entry.new(message, Severity::{{ level }}, component, time, line_number, filename)
-      end
+  def log(entry : Entry)
+    case ff = filter
+    when Filter, Proc(Entry, Bool)
+      return unless ff.call(entry)
+    when Severity
+      return unless entry.severity >= ff
```

Comment:
You mean `return if entry.severity < ff`

Reply:
"if X is inferior than Y" is simpler to understand than "unless X is superior or equal than Y", my opinion.